theatre/studio/src/panels/DetailPanel/ProjectDetails.tsx

```diff
@@ -34,41 +55,76 @@ const ProjectDetails: React.FC<{
 
   const [downloaded, setDownloaded] = useState(false)
 
-  const exportProject = useCallback(() => {
+  const exportProject = useCallback(async () => {
+    // get all possible asset ids referenced by either static props or keyframes
+    const sheets = Object.values(val(project.pointers.historic.sheetsById))
+    const staticValues = sheets
+      .flatMap((sheet) => Object.values(sheet?.staticOverrides.byObject ?? {}))
+      .flatMap((overrides) => Object.values(overrides ?? {}))
+    const keyframeValues = sheets
+      .flatMap((sheet) => Object.values(sheet?.sequence?.tracksByObject ?? {}))
+      .flatMap((tracks) => Object.values(tracks?.trackData ?? {}))
+      .flatMap((track) => track?.keyframes)
+      .map((keyframe) => keyframe?.value)
+
+    const allValues = [...staticValues, ...keyframeValues].filter(
+      // value is string, and is unique
+      (value, index, self) =>
+        typeof value === 'string' && self.indexOf(value) === index,
+    ) as string[]
+
+    const blobs = new Map<string, Blob>()
+
+    // only export assets that are referenced by the project
+    await Promise.all(
+      allValues.map(async (value) => {
+        const assetUrl = project.assetStorage.getAssetUrl(value)
+        const response = await fetch(assetUrl)
+
+        if (response.ok) {
+          blobs.set(value, await response.blob())
+        }
+      }),
+    )
+
+    if (blobs.size > 0) {
+      const zip = new JSZip()
+
+      for (const [assetID, blob] of blobs) {
```

Comment:
Since these are not actually IDs, let's use a different word here. I suggest `assetPath`

Reply:
I think it is more an ID than a path. xd It is unique like an ID, and while it represents part of the path, in itself it doesn't afford the flexibility that either absolute or relative paths do.

Reply:
Why not? Can you explain more?

Reply:
But this shouldn't hold the PR from being merged.

Reply:
It is not a path because the asset implementation right now, even with the user-managed file names, assumes the assets to be hosted as a flat directory, people cannot specify deeply nested assets. We only identify assets by their file name, which, to me, is just a special case of an ID, not really a path. But I feel like you might be advocating it for calling it a path because you have assumptions about the future implementation of this feature, in which case I'm all ears.:)

Reply:
As for a more general reasoning, I'm calling it ID, because in the end, all assets will have to have a unique id by which they can be identified. The fact that right now that ID is part of their path in the file system seems like an implementation detail that doesn't need to be alluded to in the naming.